harness/codeql-scan.yaml

```diff
@@ -0,0 +1,103 @@
+                  timeout: ""
+                  type: Run
+              - step:
+                  # OPTIONAL: Download CodeQL CLI Bundle
+                  # The CodeQL bundle (containing the CodeQL CLI as well as the pre-compiled CodeQL Query Suites, which is recommended for CI/CD integration) can either be download as par of the pipeline,
+                  # or pre-downloaded and placed on the CI/CD build machine(s). If pre-downloading, replace /path/to/cli in subsequent stages with the absolute path to the download location.
+                  # In this example, we download the latest CLI bundle (at time of writing) as part of the pipeline from https://github.com/github/codeql-action/releases.
+                  type: Run
+                  name: Download CodeQL Bundle
+                  identifier: Download_CodeQL_Bundle
+                  spec:
+                    shell: Sh
+                    command: |-
+                      sh "wget https://github.com/github/codeql-action/releases/latest/download/codeql-bundle-linux64.tar.gz -O ../codeql-bundle-linux64.tar.gz"
+                      sh "tar xzvf ../codeql-bundle-linux64.tar.gz -C ../"
+                      sh "rm ../codeql-bundle-linux64.tar.gz"
+                      sh "export PATH=$(cd ..; pwd)/codeql:$PATH"
+              - step:
+                  # Create CodeQL Database
+                  # Create a CodeQL database for a source tree that can be analyzed using one of the CodeQL products.
+                  # Note that if the --command flag is omitted for compiled languages, the AutoBuilder will be used.
+                  # Full documentation for database create step: https://codeql.github.com/docs/codeql-cli/manual/database-create/
+                  type: Run
+                  name: Create Database
+                  identifier: Create_Database
+                  spec:
+                    shell: Sh
+                    command: sh "codeql database create --language=python db"
+              - step:
+                  # Analyze CodeQL Database
+                  # Analyze a CodeQL database, producing meaningful results in the context of the source code.
+                  # Run a query suite (or some individual queries) against a CodeQL database, producing results, styled as alerts or paths, in SARIF or another interpreted format.
+                  # Note that the suite argument can accept one of the pre-compiled, out-of-the-box query suites: code-scanning, security-extended, or security-and-quality
+                  # Full documentation for database analyze step: https://codeql.github.com/docs/codeql-cli/manual/database-analyze/
+                  type: Run
+                  name: Analyze Database
+                  identifier: Analyze_Database
+                  spec:
+                    shell: Sh
+                    command: sh "codeql database analyze /codeql-dbs/repo-db javascript-security-and-quality.qls --format=sarif-latest --output=./temp/results-js.sarif"
```

Comment:
If we're running a python scan, why are we using the javascript queries?

Reply:
The way I understand this is that this workflow is more of an example of how one could use CodeQL within a Harness CI. For example, there are a couple of things hardcoded that would be good as variables to account for different commits and languages, and this workflow would only work for one language and not a list of languages. Overall, as something to get people to start using CodeQL within Harness, this seems like a good start, but I wonder if it would make more sense to include variables and a query set that matches the language being scanned.